federationsender/queue/destinationqueue.go

```diff
@@ -250,6 +250,10 @@ func (oq *destinationQueue) backgroundSend() {
 					oq.cleanPendingEDUs()
 					oq.cleanPendingInvites()
 					return
+				} else {
+					// We haven't been told to give up terminally yet
+					// but we still need to wake up the next iteration.
+					oq.wakeServerCh <- true
```

Comment:
I don't understand the implications of doing this, and the comments don't help me.

Reply:
I've updated the comments.